oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/valiation/OAuth2TokenValidatorResult.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.core.valiation;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+import org.springframework.security.oauth2.core.OAuth2Error;
+
+/**
+ * A result emitted from an {@link OAuth2TokenValidator} validation attempt
+ *
+ * @author Josh Cummings
+ * @since 5.1
+ */
+public final class OAuth2TokenValidatorResult {
+	static final OAuth2TokenValidatorResult SUCCESS = new OAuth2TokenValidatorResult(Collections.emptyList());
+
+	private final Collection<OAuth2Error> reasons;
+
+	private OAuth2TokenValidatorResult(Collection<OAuth2Error> reasons) {
+		this.reasons = new ArrayList<>(reasons);
+	}
+
+	/**
+	 * Say whether this result indicates success
+	 *
+	 * @return
+	 */
+	public boolean isSuccess() {
```

Comment:
I agree with @rwinch [comment](https://github.com/spring-projects/spring-security/pull/5624#discussion_r207608575)

> I think it is more accurate to check if there are any errors

Let's change the method name to `hasErrors`